arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTranslator.java

```diff
@@ -234,8 +234,10 @@ private String translateUnaryOp(String op, String name) {
     return name + " " + op;
   }
 
-  private static String getLiteralType(Object literal) {
-    if (literal instanceof BigDecimal) {
+  private static String getLiteralType(Object literal, RelDataType  type) {
+    if (type.getSqlTypeName() == SqlTypeName.DECIMAL) {
+      return "decimal";
+    } else if (literal instanceof BigDecimal) {
```

Comment:
since now you are passing type information to this function you should use it everywhere.  There is no reason to guess the type based on the Java type of the literal.

Reply:
done